ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -278,7 +311,9 @@ public BlobIdType getType() {
         break;
       case BLOB_ID_V2:
       case BLOB_ID_V3:
-        idBuf.put((byte) (type.ordinal() & 0x3));
+        byte flag = (byte) (type.ordinal() & 0x3);
+        flag = isEncrypted ? (byte) (flag | 0x4) : flag;
```

Comment:
minor: make the masks (0x3 and 0x4) into constants.

Reply:
This could perhaps be written as `flag |= isEncrypted ? 0x4 : 0`;